libtiff/libtiff_ctypes.py

```diff
@@ -524,24 +525,34 @@ def get_tag_define(tagname):
     @classmethod
     def open(cls, filename, mode='r'):
         """ Open tiff file as TIFF.
+
+        Parameters
+        ----------
+        filename: path-like object (str, bytes, or Path)
+          The path to the file.
+        mode: str
+          Specifies if the file is to be opened for reading ('r'), writing ('w'),
+          or appending ('a'). Optional flags can be passed. See the documentation
+          of TIFFOpen() for the complete list.
         """
-        try:
+        filename = os.fspath(filename)
+
+        if isinstance(filename, str) and hasattr(libtiff, "TIFFOpenW"):
```

Comment:
Does it make sense to always use TIFFOpenW on Windows? E.g. via
```suggestion
        if hasattr(libtiff, "TIFFOpenW"):
            if isinstance(filename, bytes):
                filename = filename.decode()
```

Reply:
We'd need to know from which encoding to decode from... and be (almost) back to doing what is done in the case of `TIFFOpen()`. I think it's fair to say if you pass bytes, we call the function that uses bytes, and if you pass unicode, we call the function that uses unicode.

Reply:
SGTM, thanks!